openhcl/underhill_attestation/src/igvm_attest/key_release.rs

```diff
@@ -162,14 +170,54 @@ pub fn parse_response(
     Ok(wrapped_key)
 }
 
+/// Convert a potentially non UTF-8 byte array into a string with non UTF-8 characters represented
+/// as hexadecimal escape sequences.
+fn string_from_utf8_preserve_invalid_bytes<T: AsRef<[u8]>>(bytes: T) -> String {
+    match std::str::from_utf8(bytes.as_ref()) {
+        Ok(utf8_str) => utf8_str.to_string(),
+        Err(err) => {
+            let (valid, invalid) = bytes.as_ref().split_at(err.valid_up_to());
+            let valid = String::from_utf8_lossy(valid).to_string();
+
+            if let Some(invalid_bytes) = err.error_len() {
+                if let Some(raw_invalid_bytes) = invalid.get(..invalid_bytes) {
+                    let raw_invalid_bytes =
+                        raw_invalid_bytes
+                            .iter()
+                            .fold(String::new(), |mut hex_string, byte| {
```

Comment:
shouldn't this fold use `valid` as its base condition?

Reply:
Resolving as the latest implementation (https://github.com/microsoft/openvmm/pull/583/commits/e3156eeddb24a549250d5e534635d881b456f4f4) removes this fold